packages/widget/src/components/Step/StepTimer.tsx

```diff
@@ -3,11 +3,62 @@ import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useTimer } from 'react-timer-hook';
 
-const getExpiryTimestamp = (step: Step) =>
-  new Date(
-    (step.execution?.process[0]?.startedAt ?? Date.now()) +
-      step.estimate.executionDuration * 1000,
-  );
+const getExpiryTimestamp = (step: Step) => {
+  const firstProcess = step.execution?.process[0];
+
+  const beginningTimestamp = firstProcess?.approvedAt ?? firstProcess?.doneAt;
```

Comment:
Why do we call it `beginningTimestamp` if it's not really the beginning of the process?